src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -197,15 +198,16 @@ static CPythonEvaluator()
                         // Reset the 'sys.path' value to the default python paths on node evaluation. 
                         var pythonNodeSetupCode = "import sys" + Environment.NewLine + "sys.path = sys.path[0:3]";
                         scope.Exec(pythonNodeSetupCode);
-
-                        ProcessAdditionalBindings(scope, bindingNames, bindingValues);
+                        var nodeName = ProcessAdditionalBindings(bindingNames, bindingValues);
 
                         int amt = Math.Min(bindingNames.Count, bindingValues.Count);
 
                         for (int i = 0; i < amt; i++)
                         {
                             scope.Set((string)bindingNames[i], InputMarshaler.Marshal(bindingValues[i]).ToPython());
                         }
+                        
+                        scope.Exec($"sys.stdout.prefix = '{nodeName}'");
```

Comment:
I guess this code could run into problems when `nodeName` contains a `'` character. Could you escape it and maybe and an `'` to a node name in the test so that we are covering it?

Reply:
i added the nodename as a variable to the PyScope, that seems to work better and works with all special characters.